greed-game/casting/artifact.py

```diff
@@ -0,0 +1,122 @@
+from casting import Actor
+
+class Artifact(Actor):
+    """A visible, moveable thing that participates in the game. 
+    
+    The responsibility of Artifact is to keep track of keep track of its kind, its value, and its visibility
+
+    Attributes:
+        _kind (string): The text that represents its kind
+        _value (int): The value it worths
+        _is_visible (boolean): if it is visible or not.
+    """
+
+    def __init__(self):
+        """Constructs a new Artifact using parent's constructor"""
+        super().__init__()
+        self._kind = "A normal one"
+        self._value = 1
+        self._is_visible = True
+
+    def get_kind(self):
+        """Gets the artifact's kind.
+        
+        Returns:
+            String: The artifact's text that represents its kind.
+        """
+        return self._kind
+
+    def get_value(self):
+        """Gets the artifact's value.
+        
+        Returns:
+            integer: The artifact's value.
+        """
+        return self._value
+
+    def get_visibility(self):
+        """Gets the artifact's visibility.
+        
+        Returns:
+            boolean: if the artifact is visible.
+        """
+        return self._is_visible
+
+    def set_kind(self, kind):
+        """Updates the kind to the given one.
+        
+        Args:
+            kind (String): The given kind.
+        """
+        self._kind = kind
+
+    def set_value(self, value):
```

Comment:
I like this part where the get and set is used to pull a value depending on the type of artifact it is